### PR TITLE
Refine reporters

### PR DESCRIPTION
**Pull request by maksymiuks**

### CODE REVIEW

File: R/check.R [hunk 1]
```diff
@@ -56,11 +60,12 @@ check_reverse_dependencies <- function(
     output = tempfile(paste(utils::packageName(), Sys.Date(), sep = "-")),
     lib.loc = .libPaths(),
     repos = getOption("repos"),
+    reverse_repos = repos,
```
Comment (maksymiuks) [inline R/check.R:63]: I added this to the high level interface to make sure users using the API can easily utilize binaries on Linux

Comment (dgkf) [reply R/check.R:63]: fine for now, though I feel like it doesn't really capture what we want the argument to do.

It's not obvious just from the argument that this implies one can use binaries and one can use source. 

We can think about what a more intuitive interface might look like. I'm imagining something like:

```r
  repos = list(source = "...", binary = "..."),
  type = c("binary", "source")  # prefer binary, use source only as fall back or when needed
```

File: R/reporter.R [hunk 2]
```diff
@@ -14,8 +14,8 @@ default_reporter <- function() {
   if (cli::is_ansi_tty()) {
     reporter("ansi_tty")
   } else if (cli::is_dynamic_tty()) {
-    stop("dynamic tty reporter not yet available")
-    reporter("dynamic_tty")
+    message("dynamic tty reporter not yet available")
```
Comment (dgkf) [inline R/reporter.R:17]:
```suggestion
```

I don't think it's worth having a message. To a typical user, it's unreasonable to assume they would know what a "dynamic_tty" is supposed to mean.

**Review by dgkf — CHANGES_REQUESTED** [review]
Some issues I encountered:

```
Warning messages:
- In task_spec(...) : partial argument match of 'package' to 'package_spec'
- In package_spec$repo : partial match of 'repo' to 'repos'
```

Please run with `options(warnPartialMatchArgs = TRUE, warnPartialMatchAttr = TRUE, warnPartialMatchDollar = TRUE)` I would recommend setting these in your `.Rprofile` generally.

Running with the two reporters seems to work fine.

**Pull request merged**